tests/test_flexible_unet.py

```diff
@@ -10,23 +10,161 @@
+    parameter_layers = [
+        [1, 1, 1, 1],
+        [2, 2, 2, 2],
+        [3, 4, 6, 3],
+        [3, 4, 6, 3],
+        [3, 4, 23, 3],
+        [3, 8, 36, 3],
+        [3, 24, 36, 3],
+    ]
+
+    def __init__(self, in_channels, pretrained, **kargs):
+        super().__init__(**kargs, n_input_channels=in_channels)
+        if pretrained:
+            # Author of paper zipped the state_dict on googledrive,
+            # so would need to download, unzip and read (2.8gb file for a ~150mb state dict).
+            # Would like to load dict from url but need somewhere to save the state dicts.
+            raise NotImplementedError(
+                "Currently not implemented. You need to manually download weights provided by the paper's author"
+                " and load then to the model with `state_dict`. See https://github.com/Tencent/MedicalNet"
+            )
+
+    @staticmethod
+    def get_inplanes():
+        return [64, 128, 256, 512]
+
+    @classmethod
+    def get_encoder_parameters(cls):
+        """
+        Get parameter list to initialize encoder networks.
+        Each parameter dict must have `spatial_dims`, `in_channels`
+        and `pretrained` parameters.
+        """
+        parameter_list = []
+        res_type: Union[ResNetBlock, ResNetBottleneck]
+        for backbone in range(len(cls.backbone_names)):
+            if backbone < 3:
+                res_type = ResNetBlock
+            else:
+                res_type = ResNetBottleneck
+            parameter_list.append(
+                {
+                    "block": res_type,
+                    "layers": cls.parameter_layers[backbone],
+                    "block_inplanes": ResNetEncoder.get_inplanes(),
+                    "spatial_dims": 2,
+                    "in_channels": 3,
+                    "pretrained": False,
+                }
+            )
+        return parameter_list
+
+    @classmethod
+    def get_output_feature_channels(cls):
+        """
+        Get number of output features' channel.
+        """
+        return cls.output_feature_channels
+
+    @classmethod
+    def get_output_feature_numbers(cls):
+        """
+        Get number of output feature.
+        """
+        return [4] * 7
+
+    @classmethod
+    def get_encoder_names(cls):
+        """
+        Get the name string of backbones which will be used to initialize flexible unet.
+        """
+        return cls.backbone_names
+
+    def forward(self, x: torch.Tensor):
+        feature_list = []
+        x = self.conv1(x)
+        x = self.bn1(x)
+        x = self.relu(x)
+        if not self.no_max_pool:
+            x = self.maxpool(x)
+        x = self.layer1(x)
+        feature_list.append(x)
+        x = self.layer2(x)
+        feature_list.append(x)
+        x = self.layer3(x)
+        feature_list.append(x)
+        x = self.layer4(x)
+        feature_list.append(x)
+
+        return feature_list
+
+
+BACKBONE.regist_class(ResNetEncoder)
```

Comment:
Could you please help add a test case to ensure the `FlexibleUNet` with registered `ResNetEncoder` can do `forward()` computation?

Thanks in advance.

Reply:
Hi @Nic-Ma ,
      I have added the resnet test in [test cases] 
(https://github.com/binliunls/MONAI/blob/d981093403bdad9abfca5ddbbb9c8035ba4e9664/tests/test_flexible_unet.py#L218) and will update the comment. I think this test should be able to verify forward function of  resnet encoder.
Thanks,
Bin